controllers/helper.go

```diff
@@ -19,24 +19,28 @@ package controllers
 import (
 	"context"
 
+	corev1 "k8s.io/api/core/v1"
+
 	kciv1alpha1 "github.com/kloeckner-i/db-operator/api/v1alpha1"
 	"github.com/kloeckner-i/db-operator/pkg/utils/kci"
 	crdv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 )
 
-func isDBSpecChanged(dbcr *kciv1alpha1.Database) bool {
+func isDBSpecChanged(dbcr *kciv1alpha1.Database, databaseSecret *corev1.Secret) bool {
 	annotations := dbcr.ObjectMeta.GetAnnotations()
 
-	return annotations["checksum/spec"] != kci.GenerateChecksum(dbcr.Spec)
+	return annotations["checksum/spec"] != kci.GenerateChecksum(dbcr.Spec) ||
+		     annotations["checksum/secret"] != kci.GenerateChecksum(databaseSecret.Data)
 }
 
-func addDBSpecChecksum(dbcr *kciv1alpha1.Database) {
+func addDBSpecChecksum(dbcr *kciv1alpha1.Database, databaseSecret *corev1.Secret) {
```

Comment:
maybe it's also better to be renamed here to `addDBChecksum`

Reply:
done, renamed to `addDBChecksum`